src/App.tsx

```diff
@@ -8,52 +8,46 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
 import { Post } from './types/Post';
-import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { loadApiPosts, setPosts } from './features/postsSlice';
+import { loadApiUsers } from './features/usersSlice';
+import { setPost } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const { items, hasError, loaded } = useAppSelector(state => state.posts);
+  const author = useAppSelector(state => state.author.author);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const selectedPost = useAppSelector(state => state.selectedPost.post);
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
+  const setSelectedPost = (post: Post | null) => {
+    dispatch(setPost(post));
```

Comment:
Consider using a useCallback for the setSelectedPost function to prevent unnecessary re-renders. 😇